certbot/compat/filesystem.py

```diff
@@ -36,6 +43,63 @@ def chmod(file_path, mode):
         _apply_win_mode(file_path, mode)
 
 
+def makedirs(file_path, mode=0o777, exists_ok=False):
+    # type: (str, int, bool) -> None
+    """
+    Rewrite of original os.makedirs function, that will ensure on Windows that given mode
+    is correctly applied.
+    :param str file_path: The file path to open
+    :param int mode: POSIX mode to apply on leaf directory when created, Python defaults
+                     will be applied if ``None``
+    :param bool exists_ok: If set to ``True``, do not raise exception if leaf directory
+                           already exists.
+    """
+    orig_mkdir_fn = os.mkdir
+    try:
+        # As we know that os.mkdir is called internally by os.makedirs, we will swap the function in
+        # os module for the time of makedirs execution on Windows.
+        os.mkdir = mkdir if not POSIX_MODE else os.mkdir  # type: ignore
```

Comment:
`if not POSIX_MODE else os.mkdir` is doubly unnecessary if `mkdir` has its own `if POSIX_MODE`, and if we bail early as I suggest above.

Reply:
See https://github.com/certbot/certbot/pull/6968#issuecomment-506988582